ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-traffic-engineering.j2

```diff
@@ -0,0 +1,43 @@
+{% if router_traffic_engineering is arista.avd.defined %}
+
+## Router Traffic-Engineering
+{%     if router_traffic_engineering.segment_routing is arista.avd.defined %}
+
+### Segment Routing Summary
+
+- SRTE is enabled.
+{%         if router_traffic_engineering.segment_routing.colored_tunnel_rib is arista.avd.defined(true) %}
+
+- system-colored-tunnel-rib is enabled
+{%         endif %}
+{%         if router_traffic_engineering.segment_routing.policy_endpoints is arista.avd.defined %}
+
+#### SRTE Policies
+
+| Endpoint | Color | Preference | Name | Description | Label Stack | Index  | Weight | Explicit Null |
+| -------- | ----- | ---------- | ---- | ----------- | ----------- | ------ | ------ | ------------- |
+{%             for endpoint in router_traffic_engineering.segment_routing.policy_endpoints | arista.avd.natural_sort %}
+{%                 for color in endpoint.colors | arista.avd.natural_sort %}
+{%                     set name = color.name | arista.avd.default('-') %}
+{%                     set description = color.description | arista.avd.default('-') %}
+{%                     for pathgroup in color.path_group | arista.avd.natural_sort %}
+{%                         set pathgroup_preference = pathgroup.preference | arista.avd.default('-') %}
+{%                         set expnull = pathgroup.explicit_null | arista.avd.default('-') %}
+{%                         for labelstack in pathgroup.segment_list | arista.avd.natural_sort %}
+{%                             set stack = labelstack.label_stack %}
+{%                             set index = labelstack.index | arista.avd.default('-') %}
+{%                             set weight = labelstack.weight | arista.avd.default('-') %}
+| {{ endpoint.address }} | {{ color.value }} | {{ pathgroup_preference }} | {{ name }} | {{ description }} | {{ stack }} | {{ index }} | {{ weight }} | {{ expnull }} |
```

Comment:
I see there is a problem with this documentation. If segment_list is not defined but path_group and color are defined. We won't see that entry.

We need a work around to solve this.

Reply:
We need to place this line 30 after line33 (endfor). And handle the default values at the same time.

Reply:
The minimal config that I could configure was ```policy endpoint 10.10.1.1 color 234```. So I would place line 30 after line32 (endfor). This ensures the documentation is done for this minimal config as well.

Reply:
But the above solution imposes another problem. For example label_stack is not defined. So the var **stack** is not generated.

So we need to use ```stack | arista.avd.default('-')```.

Like:

```
| {{ endpoint.address }} | {{ color.value }} | {{ pathgroup_preference | arista.avd.default('-') }} | {{ name }} | {{ description }} | {{ stack | arista.avd.default('-') }} | {{ index | arista.avd.default('-') }} | {{ weight | arista.avd.default('-') }} | {{ expnull | arista.avd.default('-') }} |
```

Reply:
Never mind. That would not work. The current model is good.

Reply:
Additional check and create empty segment-list could be possible after line 25:

```Jinja2
{%                         if pathgroup.segment_list is not arista.avd.defined %}
{%                             set pathgroup = {'segment_list': [{ 'label_stack': '-'}]} %}
{%                         endif %}
```

What do you think?